tests/test_backend.py

```diff
@@ -114,27 +114,29 @@ def find_root_full(a):
 @pytest.mark.unit
 def test_lstsq():
     """Test cholesky factorization of least squares solution."""
+    rng = np.random.default_rng(seed=0)
```

Comment:
This test ran fine for me locally and on della, but would fail on CI.

Because the seed wasn't fixed before, the particular values would depend on which other tests were run and in what order, etc so its hard to debug. Manually fixing the seed seems to fix the issue (likely the rng was just spitting out some weird values leading to ill-conditioned matrices)